SimTKcommon/Simulation/include/SimTKcommon/internal/MeasureImplementation.h

```diff
@@ -1289,6 +1289,43 @@ template <class T>
 class Measure_Differentiate_Result {
 public:
     Measure_Differentiate_Result() : derivIsGood(false) {}
+
+    Xml::Element toXmlElement(const std::string& name) const {
+        static const int version = 1;
+        Xml::Element e("Measure_Differentiate_Result");
+        if (!name.empty()) e.setAttributeValue("name", name);
+        e.setAttributeValue("version", String(version));
+        e.appendNode(toXmlElementHelper(operand, "operand", true));
+        e.appendNode(toXmlElementHelper(operandDot, "operandDot", true));
+        e.appendNode(toXmlElementHelper(derivIsGood, "derivIsGood", true));
```

Comment:
Yeah, the "Helper" method should just be needed in the internal implementation. Here it is appearing in user-written code for serializing. 

The issue is that we normally want classes to have `toXmlElement()` methods. But when serializing an existing class (`std::vector`, say) you can't add a new method. So you have to write a free function `toXmlElement(std::vector<T>)` to serialize it. What if there is both a member named `MyClass::toXmlElement()` and a free function `toXmlElement(MyClass)`? I wanted to ensure that the member gets called. By template resolution rules, that `true` value is a better match for a `bool` than an `int`. So I arranged things so that the member function gets called from a Helper that takes a `bool` while the free function gets called from a Helper that takes an `int`. That one gets called if there's no `bool` competing with it because `true` gets promoted to `int`.

All that is still necessary; I'm just thinking it doesn't have to appear in user code. Use code could call some nicely-named method, then that method could invoke the Helper which would then pick out the right member or free function. I *think* that should be possible ...